pkg/cri/sbserver/image_remove.go

```diff
@@ -33,15 +35,18 @@ import (
 // Remove the whole image no matter the it's image id or reference. This is the
 // semantic defined in CRI now.
 func (c *criService) RemoveImage(ctx context.Context, r *runtime.RemoveImageRequest) (*runtime.RemoveImageResponse, error) {
+	ctx, span := tracing.StartSpan(ctx, makeSpanName("RemoveImage"))
+	defer span.End()
 	image, err := c.localResolve(r.GetImage().GetImage())
 	if err != nil {
+		span.RecordError(err)
 		if errdefs.IsNotFound(err) {
 			// return empty without error when image not found.
 			return &runtime.RemoveImageResponse{}, nil
 		}
 		return nil, fmt.Errorf("can not resolve %q locally: %w", r.GetImage().GetImage(), err)
 	}
-
+	span.SetAttributes(attribute.String("image.ref", image.ID))
```

Comment:
We should pick either `image.name` or `image.ref`, assuming both of them would have something like `docker.io/library/debian:latest`.

Reply:
Wherever I used image.name refers to name of the images `docker.io/library/debian:latest`

 image.ref refers to the image id (which is the digest of the image config) `sha256:51086ed63d8cba3a6a3d94ecd103e9638b4cb8533bb896caf2cda04fb79b862f`

Reply:
Is the digest same as the image's digest? How about calling it `image.digest`

We use `ref` as a way to refer the human-readable image name. We shouldn't use that for something else.

https://github.com/containerd/containerd/blob/badb66113cceb480b6c37fd18adae400f309c57c/cmd/ctr/commands/images/images.go#L94

Reply:
I see, sure I can change `image.name` to `image.ref` 

As for `ctr`
```
[ec2-user@ip-172-31-62-162 client]$ sudo ctr i ls
REF                             TYPE                                                      DIGEST                                                                  SIZE     PLATFORMS                                                                                               LABELS
docker.io/library/debian:latest application/vnd.docker.distribution.manifest.list.v2+json sha256:e538a2f0566efc44db21503277c7312a142f4d0dedc5d2886932b92626104bff 52.5 MiB linux/386,linux/amd64,linux/arm/v5,linux/arm/v7,linux/arm64/v8,linux/mips64le,linux/ppc64le,linux/s390x -
```

`Digest` here refers to the image Index file,

```
[ec2-user@ip-172-31-62-162 client]$ cat  /var/lib/containerd/io.containerd.content.v1.content/blobs/sha256/e538a2f0566efc44db21503277c7312a142f4d0dedc5d2886932b92626104bff| json_reformat | tail -n 10
            "platform": {
                "architecture": "s390x",
                "os": "linux"
            },
            "size": 529
        }
    ],
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "schemaVersion": 2
}
```

this is different from `image.id` which refers to the image config digest.

The way I see this, `ID`  is used to refer to unique images (platform + name), `REF` is a generic name for the image that can belong to any platform, `DIGEST` is used to denote the index file for the image.

Reply:
Okay, this is what I am using now:
`image.ref` - name of the image such as docker.io/library/debian:latest
`image.id` - digest of the image config which is also the unique image identifier